src/main/java/com/cloudbees/jenkins/support/slowrequest/InflightRequest.java

```diff
@@ -36,14 +40,27 @@
      */
     File record;
 
+    /**
+     * Username of user who made the http call.
+     */
+    final String userName;
+
+    /**
+     * Referer link to track any redirect urls.
+     */
+    final String referer;
 
     InflightRequest(HttpServletRequest req) {
         String query = req.getQueryString();
         url = req.getRequestURL() + (query == null ? "" : "?" + query);
         startTime = System.currentTimeMillis();
+        userName = Jenkins.getAuthentication().getName();
+        referer = ((StaplerRequest)req).getReferer();
```

Comment:
Unsafe cast. I would recommend to check the class and to default the field to something if it is not <code> StaplerRequest</code>